### PR TITLE
Integrate postgres into realm server

### PR DESCRIPTION
**Pull request by habdelra**
This PR enables postgres in the app :tada: 

When you start your servers you can use:
```
PG_INDEXER=true start:all
```
to start the servers with postgres. Everything should be automatic--the DB will migrate automatically as part of starting the server. This introduces both the postgres based index as well as our postgres based queuing system.

you can connect to your DB using psql:
```sh
psql -h localhost -p 5435 -U postgres
```

### CODE REVIEW

File: packages/realm-server/pg-queue.ts [hunk 1]
```diff
@@ -112,13 +112,14 @@ export default class PgQueue implements Queue {
...
 
   private pollInterval = 10000;
-  private pgClient = new PgAdapter();
   private handlers: Map<string, Function> = new Map();
   private notifiers: Map<number, Deferred<any>> = new Map();
 
   private jobRunner: WorkLoop | undefined;
   private notificationRunner: WorkLoop | undefined;
 
+  constructor(private pgClient: PgAdapter) {}
```
Comment (backspace) [inline packages/realm-server/pg-queue.ts:121]: I didn’t know this existed to set the property 😯

File: packages/realm-server/scripts/start-base-root.sh [hunk 2]
```diff
@@ -1,6 +1,21 @@
 #! /bin/sh
+check_postgres_ready() {
+  docker exec boxel-pg pg_isready -U postgres >/dev/null 2>&1
+}
+# remove this check after the feature flag is removed
+if [ -n "$PG_INDEXER" ]; then
+  while ! check_postgres_ready; do
+    printf '.'
+    sleep 1
```
Comment (backspace) [inline packages/realm-server/scripts/start-base-root.sh]: Should there be a timeout like I’m working on adding [here](https://github.com/cardstack/boxel/blob/825d4511aa504d04c0256ddd6e52b37cf2ee8c69/packages/matrix/scripts/register-realm-users.sh#L12)?

File: packages/realm-server/scripts/start-drafts-root.sh [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+check_postgres_ready() {
+  docker exec boxel-pg pg_isready -U postgres >/dev/null 2>&1
+}
+# remove this check after the feature flag is removed
+if [ -n "$PG_INDEXER" ]; then
+  while ! check_postgres_ready; do
+    printf '.'
+    sleep 1
+  done
+fi
```
Comment (backspace) [inline packages/realm-server/scripts/start-drafts-root.sh]: Since this is repeated across a few scripts, maybe it should be extracted?

**Review by backspace — APPROVED** [review]
boxel_index — boxel_dev — boxel realm server 2024-05-02 16-31-06

nice 🤩

File: packages/realm-server/scripts/start-base-root.sh (hunk 2 above)
Comment (habdelra) [reply packages/realm-server/scripts/start-base-root.sh]: Sure

File: packages/realm-server/scripts/start-drafts-root.sh (hunk 3 above)
Comment (habdelra) [reply packages/realm-server/scripts/start-drafts-root.sh]: Sounds good

File: packages/host/tests/integration/components/operator-mode-test.gts [hunk 4]
```diff
@@ -3654,7 +3654,9 @@ module('Integration | operator-mode', function (hooks) {
     assert
       .dom('[data-test-card-url-bar-error]')
       .containsText('This resource does not exist');
-    await percySnapshot(assert);
+    // Percy is failing to capture this snapshot for some
+    // reason. creating issue for this CS-6780
+    // await percySnapshot(assert);
```
Comment (jurgenwerk) [inline packages/host/tests/integration/components/operator-mode-test.gts:3659]: Already tracking this in [CS-6767](https://linear.app/cardstack/issue/CS-6767/investigate-percy-timeouts)

**Pull request merged**